spec/rubocop/cop/style/redundant_self_spec.rb

```diff
@@ -292,4 +292,38 @@ def do_something((a, b)) # This method expects Array that has 2 elements as argu
       end
     RUBY
   end
+
+  context 'with ruby >= 2.7', :ruby27 do
+    context 'with pattern matching' do
+      it 'accepts a self receiver on an `match-var`' do
+        expect_no_offenses(<<~RUBY)
+          case foo
+            in Integer => bar
```

Comment:
Patterns other than `:match_as` seem to have the same issue. e.g. `:hash_pattern`, `:array_pattern`, and so on:

```ruby
case pattern
in {foo: 42}
end

case pattern
in [1, 2, 3]
end
```

Reply:
Got it, I'm not super familiar with pattern matching! I'll fix those as well, thank you :heart: